modules/programs/fff.nix

```diff
@@ -0,0 +1,582 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.fff;
+
+in {
+  meta.maintainers = [ maintainers.gigahawk ];
+
+  options.programs.fff = {
+    enable = mkEnableOption "A simple file manager written in bash";
+
+    cdOnExit = {
+      enable = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Change directory to current folder on exit
+        '';
+      };
+
+      name = mkOption {
+        type = types.str;
+        default = "f";
+        description = ''
+          Name of function used to call fff.
+        '';
+      };
+
+      commands = mkOption {
+        type = types.str;
+        default = ''
+          # cd to exited folder for bash and zsh
+          fff "$@"
+          cd "$(cat "''${XDG_CACHE_HOME:=''${HOME}/.cache}/fff/.fff_d")" 
+        '';
+        example = "";
+        description = ''
+          Bash/Zsh commands used to start fff with cdOnExit enabled.
+        '';
+      };
+
+      fishCommands = mkOption {
+        type = types.str;
+        default = ''
+          fff $argv
+          cd (cat ${cfg.cdOnExit.helperFile})
+        '';
+        description = ''
+          Fish commands used to start fff with cdOnExit enabled.
+        '';
+      };
+
+      helperFile = mkOption {
+        type = types.str;
+        default = "${config.home.homeDirectory}/.cache/fff/.fff_d";
+        description = ''
+          Location of CD on exit helper file
+        '';
+      };
+    };
+
+    config = {
+      lsColors = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Use LS_COLORS to color fff
+        '';
+      };
+      color1 = mkOption {
+        type = types.ints.between 0 9;
+        default = 2;
+        description = ''
+          Directory color [0-9]
+        '';
+      };
+      color2 = mkOption {
+        type = types.ints.between 0 9;
+        default = 7;
+        description = ''
+          Status color [0-9]
+        '';
+      };
+      color3 = mkOption {
+        type = types.ints.between 0 9;
+        default = 6;
+        description = ''
+          Selection color [0-9] (copied/moved files)
+        '';
+      };
+      color4 = mkOption {
+        type = types.ints.between 0 9;
+        default = 1;
+        description = ''
+          Cursor color [0-9]
+        '';
+      };
+      fileOpener = mkOption {
+        type = types.str;
+        default = "xdg-open";
+        description = ''
+          File opener
+        '';
+      };
+      trash = mkOption {
+        type = types.str;
+        default = "${config.home.homeDirectory}/.local/share/fff/trash";
+        description = ''
+          Path to trash directory
+        '';
+      };
+      trashCmd = mkOption {
+        type = types.str;
+        default = "mv";
+        description = ''
+          Command to use when trashing a file
+        '';
+      };
+      favorite1 = mkOption {
```

Comment:
Use list or attribute set instead of explicit numbers in option name.

Reply:
User's could conceivably want to set favorites on a specific number without setting ones prior, so an attribute set was used.